src/tiledb/cloud/dag/dag.py

```diff
@@ -1801,7 +1862,56 @@ def _update_status(self) -> None:
                     futures.execute_callbacks(nd, cbs)
                 with self._lifecycle_condition:
                     self._set_status(Status.FAILED)
-                raise  # Bail out and fail loudly.
+                raise  # Bail out and fail loudly
+
+    def _tdb_to_json(self, override_name: Optional[str] = None) -> Dict[str, Any]:
+        """Converts this DAG to a registerable format.
+
+        :param override_name: Name to override DAG conversion.
+        :return: Mapping of DAG tree to json for submission to REST.
+        """
+
+        nodes = _topo_sort_nodes(self.nodes)
+
+        existing_names = set(self.nodes_by_name.keys())
+
+        node_jsons = [n.to_registration_json(existing_names) for n in nodes]
+
+        for n, n_json in zip(nodes, node_jsons):
+            n_json["depends_on"] = [str(parent.id) for _, parent in n.parents.items()]
+
+        return {
+            "name": override_name or self.name,
+            "nodes": node_jsons,
+        }
+
+    def register(
+        self,
+        override_name: Optional[str] = None,
+    ) -> str:
+        """Register DAG to TileDB.
+
+        :param override_name: Name to register DAG as. Uses self.name as default.
+        :return: Registered name of task graph.
+        """
+
+        tg_name = override_name or self.name
+
+        if not tg_name:
+            raise ValueError(
+                "Must specify registration name to DAG.name or override_name."
+            )
+
+        registration.register(
+            graph=self,
+            name=tg_name,
+            namespace=self.namespace,
+        )
+
+        with tiledb.open(f"tiledb://{self.namespace}/{tg_name}", "w") as A:
+            A.meta["dataset_type"] = "registered_task_graph"
```

Comment:
It would be nice if the registration endpoint could take metadata, then this would be more atomic.